pl_examples/hydra_examples/pl_template.py

```diff
@@ -0,0 +1,48 @@
+"""
+Pytorch Lightning training using Hydra for configuration
+"""
+
+import hydra
+from omegaconf import DictConfig
+from pl_examples.hydra_examples.user_config import conf_init
+
+from pl_examples.models.hydra_config_model import LightningTemplateModel
+from pytorch_lightning import Callback, seed_everything, Trainer
+
+
+@hydra.main(config_path="conf", config_name="config")
+def main(cfg: DictConfig):
+    """
+    Main training routine specific for this project
+    :param cfg:
+    """
+    seed_everything(cfg.model.seed)
+    # ------------------------
+    # 1 INIT LIGHTNING MODEL
+    # ------------------------
+    model = LightningTemplateModel(cfg)
+    # ------------------------
+    # 2 INIT TRAINER
+    # ------------------------
+
+    callbacks = (
+        [hydra.utils.instantiate(c) for c in cfg.callbacks.callbacks_list] if "callbacks_list" in cfg.callbacks else []
+    )
+
+    trainer = Trainer(
+        **cfg.trainer,
```

Comment:
For example this line, would be nice to accept a DictConfig

Reply:
I think this can be a part of the discussion. I am not sure if it's possible to do this in a backward compatible way.